src/services/paras/ParasService.ts

```diff
@@ -223,18 +223,45 @@ export class ParasService extends AbstractService {
 			phase: IOption<AuctionPhase>,
 			winning;
 		if (auctionInfoOpt.isSome) {
+			/**
+			 * AuctionInfo:::<T>:get() when Some, it returns a tuple where the first item is the
+			 * lease period index that the first of the four contiguous lease periods
+			 * on auction is for. The second is the block number when the auction will
+			 * 'being to end', i.e. the first block of the Ending Period of the auction
+			 */
 			[leasePeriodIndex, beginEnd] = auctionInfoOpt.unwrap();
-			const endingOffset = this.endingOffset(blockNumber, beginEnd);
+
+			/**
+			 * End of current auctions endPeriod.
+			 */
+			finishEnd = beginEnd.add(endingPeriod);
+			/**
+			 * We determine what our phase is so we can decide how to calculate our
+			 * ending offset.
+			 */
+			if (finishEnd.lt(blockNumber)) {
+				phase = 'vrfDelay';
+			} else {
+				phase = beginEnd.gt(blockNumber) ? 'startPeriod' : 'endPeriod';
+			}
+
+			/**
+			 * The endingOffset according to polkadot has two potential phases
+			 * where this will be a valid block number. Both `startPeriod`, and `endPeriod`
+			 * have valid offsets.
+			 */
+			const endingOffset = this.endingOffset(
+				blockNumber,
+				beginEnd,
+				phase,
+				historicApi
+			);
 
 			const winningOpt = endingOffset
 				? await historicApi.query.auctions.winning<Option<WinningData>>(
 						endingOffset
 				  )
-				: await historicApi.query.auctions.winning<Option<WinningData>>(
-						// when we are not in the ending phase of the auction winning bids are stored at 0
-						0
-				  );
-
+				: { isSome: undefined };
 			if (winningOpt.isSome) {
```

Comment:
```suggestion
			if (endingOffset) {
			    let winningOffset = await historicApi.query.auctions.winning<Option<WinningData>>(
						endingOffset
				  );
			
```

Reply:
If understand correctly, these should be the equivalent logic of what you have, but a little simpler

Reply:
Nice refactor! :), yea like the change a lot